spec/src/main/resources/META-INF/smithy/bsp/bsp.smithy

```diff
@@ -1150,12 +1146,20 @@ structure CompileParams {
 
     /// Optional arguments to the compilation process.
     arguments: Arguments
+
+    /// Optional environment variables to set before compiling.
+    environmentVariables: EnvironmentVariables
```

Comment:
(I don't think that can easily be supported by sbt or mill)

Reply:
I wonder why environment variables are needed at all when compiling things? Compilers aren't typically configured via env vars. Most build tools don't have dedicated support for env vars that are passed to the compiler. And without such support, depending on env vars in the compilation would make it less reproducible. What is the motivation for that addition?

Reply:
> I wonder why environment variables are needed at all when compiling things? 

In most cases - they are not needed.

> Compilers aren't typically configured via env vars. 

Agree. But in some cases - they can be. For example: make can read env variables. In Bazel you can also [use env variables in the build process](https://bazel.build/reference/command-line-reference#flag--action_env). 

> What is the motivation for that addition?

Futureproofing. Ofc sbt & mill are free to ignore these arguments.

Reply:
Yeah, in Mill you can also use env variables (and you can even track their value to capture any change), but I think that is not the point. The question is, do you need control over "different" env vars per compiler invocation? Where should these env vars come from? Do you expect special configuration for it in the BSP client? If it's a special configuration thing inside the build tools, shouldn't it be completely transparent to the client? 

We are not speaking about general stable env vars set in the environment of the user and staying the same while the BSP client is running, right?

Reply:
> Where should these env vars come from? Do you expect special configuration for it in the BSP client?

Yes, I imagine you could have some UI for it in a client

> We are not speaking about general stable env vars set in the environment of the user and staying the same while the BSP client is running, right?

Right. This would be used for overrides only.

Reply:
But in general I'm not particularly attached to this specific addition, as I said I added it for future-proofing and consistency with other requests. But we have no plans to implement compile-specific env var settings in bazel-bsp. So I'm 100% fine with cutting this out and revisiting it if anyone needs it later.

Reply:
My guess, this is the kind of future-proofing that produces more work and complexity, but never pays out. But you it appears to me you have a real use-case in mind, so I'll mute. The spec should be clear whether this feature is optional or not for BSP servers, as it makes compilation tasks quasi uncachable.

Reply:
I'll just remove this part and if we decide to implement it I'll come back :)